tools/android/packaging/xbmc/src/org/xbmc/kodi/XBMCJsonRPC.java.in

```diff
@@ -0,0 +1,647 @@
+        e.printStackTrace();
+        return null;
+      }
+    } catch (Exception e)
+    {
+      Log.e(TAG, "Failed to read JSON");
+      e.printStackTrace();
+      return null;
+    }
+  }
+
+  public JSONObject request_object(String jsonRequest)
+  {
+    try
+    {
+      JSONObject resp = new JSONObject(request_string(jsonRequest));
+      return resp;
+    } catch (Exception e)
+    {
+      Log.e(TAG, "Failed to parse JSON");
+      e.printStackTrace();
+      return null;
+    }
+  }
+
+  public JSONArray request_array(String jsonRequest)
+  {
+    try
+    {
+      JSONArray resp = new JSONArray(request_string(jsonRequest));
+      return resp;
+    } catch (Exception e)
+    {
+      Log.e(TAG, "Failed to parse JSON");
+      e.printStackTrace();
+      return null;
+    }
+  }
+
+  public Bitmap getBitmap(String src)
+  {
+    if (src.isEmpty())
+      return null;
+
+    try
+    {
+      JSONObject req = request_object("{\"jsonrpc\": \"2.0\", \"method\": \"Files.PrepareDownload\", \"params\": { \"path\": \""
+          + src + "\"}, \"id\": \"1\"}");
+      if (req == null)
+        return null;
+
+      JSONObject result = req.getJSONObject("result");
+      String surl = result.getJSONObject("details").getString("path");
+
+      URL url = new URL(m_jsonURL + "/" + surl);
+      HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+      connection.setDoInput(true);
+      connection.connect();
+      InputStream input = connection.getInputStream();
+      Bitmap myBitmap = BitmapFactory.decodeStream(input);
+      return myBitmap;
+    } catch (Exception e)
+    {
+      e.printStackTrace();
+      return null;
+    }
+  }
+
+  public String getBitmapUrl(String src)
+  {
+    if (src.isEmpty())
+      return null;
+
+    try
+    {
+      JSONObject req = request_object("{\"jsonrpc\": \"2.0\", \"method\": \"Files.PrepareDownload\", \"params\": { \"path\": \""
+          + src + "\"}, \"id\": \"1\"}");
+      if (req == null)
+        return null;
+
+      JSONObject result = req.getJSONObject("result");
+      String surl = result.getJSONObject("details").getString("path");
+
+      return (m_jsonURL + "/" + surl);
+    } catch (Exception e)
+    {
+      e.printStackTrace();
+      return null;
+    }
+  }
+
+  public Cursor search(String query)
+  {
+      String[] menuCols = new String[] {
+              BaseColumns._ID,
+              COLUMN_TITLE,
+              COLUMN_TAGLINE,
+              COLUMN_THUMB,
+              COLUMN_FANART,
+      };
+      MatrixCursor mc = new MatrixCursor(menuCols);
+
+      try
+      {
+        JSONObject req = request_object(String.format(SEARCH_MOVIES_JSON, /*"\"operator\": \"contains\", \"field\": \"title\", \"value\": \"" + query + "\"", limit));*/
+        "\"or\": [" +
+        "{\"operator\": \"contains\", \"field\": \"title\", \"value\": \"" + query + "\"}," +
+        "{\"operator\": \"contains\", \"field\": \"set\", \"value\": \"" + query + "\"}," +
```

Comment:
the reason why I suggested tagline was that regardless how hard I tried pronouncing a specific movie title, I never got a match because Google does a poor job with detecting English words when voice search is set f.e. to German. And IIRC that movie had an English name also in German, so the only way to get a match was the tagline, which worked fine with Google Play Movies. Once I switched voice search language to English, movie was detected just fine, but then none of the German titled movies where found.